### PR TITLE
[prometheus-monitors] subchart for Pod|ServiceMonitors with default relabelings

### PR DESCRIPTION
**Pull request by IvoGoman**
For proper alert routing we require certain labels to be attached to a metric (e.g. region, cluster_type). Many Pod|ServiceMonitors in this repository do not set these labels. The problem is that every chart maintainer must be aware of these mandatory labels for alert routing.
Previously this was not an issue, since most pods/services were scraped by one scrape job discovering targets via the prometheus scrape annotations.
This Subchart should provide a similarly convenient  option of adding a Pod|ServiceMonitor with a certain set of defaults and the option to further adjust the scrape configuration.

Example metric with default relabelings:
https://prometheus-infra-collector.qa-de-1.cloud.sap/graph?g0.expr=metis_replication_restart_counter%7Bpod%3D~%22mariadb-sync-nova-api.*%22%7D&g0.tab=1&g0.stacked=0&g0.show_exemplars=0&g0.range_input=1h

Example PR for basic usage: #4337

### CODE REVIEW

File: common/prometheus-monitors/templates/podmonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{{- if .Values.podMonitor.enabled }}
+{{- if .Values.serviceMonitor.enabled }}
+{{ fail "failed to render PodMonitor. ServiceMonitor is also enabed. Only one is allowed."}}
+{{- end }}
```
Comment (richardtief) [inline common/prometheus-monitors/templates/podmonitor.yaml:4]:
```suggestion
{{- if and .Values.podMonitor.enabled .Values.serviceMonitor.enabled }}
{{ fail "Could not render Podmonitor. '.Values.podMonitor.enabled' and '.Values.serviceMonitor.enabled' are mutually exclusive."}}
{{- end }}
```

File: common/prometheus-monitors/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{{- if .Values.serviceMonitor.enabled }}
+{{- if .Values.podMonitor.enabled }}
+{{ fail "failed to render ServiceMonitor. PodMonitor is also enabed. Only one is allowed."}}
+{{- end }}
```
Comment (richardtief) [inline common/prometheus-monitors/templates/servicemonitor.yaml:4]: {{- if and .Values.podMonitor.enabled .Values.serviceMonitor.enabled }}
{{ fail "Could not render Servicemonitor. '.Values.podMonitor.enabled' and '.Values.serviceMonitor.enabled' are mutually exclusive."}}
{{- end }}

File: common/prometheus-monitors/values.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  clusterType: controlplane
+
+podMonitor:
+  enabled: false
+serviceMonitor:
+  enabled: false
+
+prometheus: ""
+
+namespaces: []
```
Comment (richardtief) [inline common/prometheus-monitors/values.yaml:13]:
```suggestion
# List of namespace names from which the Endpoint objects are selected.
namespaces: []
```

File: common/prometheus-monitors/values.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+matchExpressions: []
```
Comment (richardtief) [inline common/prometheus-monitors/values.yaml:15]:
```suggestion
# matchExpressions is a list of label selector requirements. The requirements are ANDed.
matchExpressions: []
```

File: common/prometheus-monitors/values.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+matchLabels: []
```
Comment (richardtief) [inline common/prometheus-monitors/values.yaml:16]:
```suggestion

# matchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels map
# is equivalent to an element of matchExpressions, whose key field is "key", the operator
# is "In", and the values array contains only "value". The requirements are ANDed.
matchLabels: []
```

File: common/prometheus-monitors/values.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+customRelabelings: []
+customMetricRelabelings: []
```
Comment (richardtief) [inline common/prometheus-monitors/values.yaml:19]:
```suggestion

# RelabelConfig allows dynamic rewriting of the label set, being applied to
# samples before ingestion.
customMetricRelabelings: []
```

File: common/prometheus-monitors/values.yaml [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  enabled: false
+
+prometheus: ""
+
+namespaces: []
+
+matchExpressions: []
+matchLabels: []
+
+customRelabelings: []
```
Comment (richardtief) [inline common/prometheus-monitors/values.yaml:18]:
```suggestion
#  RelabelConfigs to apply to samples before scraping.
customRelabelings: []
```

**Review by richardtief — CHANGES_REQUESTED** [review]
Thank you for taking this on. I just have some suggestions like some descriptions on the values.

File: common/prometheus-monitors/templates/podmonitor.yaml (hunk 1 above)
Comment (IvoGoman) [reply common/prometheus-monitors/templates/podmonitor.yaml:4]: This is intentionally split. The outer if is for determining if the template should be rendered the inner if to fail if both service and pod monitor are enabled.

File: common/prometheus-monitors/templates/servicemonitor.yaml (hunk 2 above)
Comment (IvoGoman) [reply common/prometheus-monitors/templates/servicemonitor.yaml:4]: This is intentionally split. The outer if is for determining if the template should be rendered the inner if to fail if both service and pod monitor are enabled.

File: common/prometheus-monitors/templates/podmonitor.yaml (hunk 1 above)
Comment (richardtief) [reply common/prometheus-monitors/templates/podmonitor.yaml:4]: I see. Overlooked it at first glance.

File: common/prometheus-monitors/templates/servicemonitor.yaml (hunk 2 above)
Comment (richardtief) [reply common/prometheus-monitors/templates/servicemonitor.yaml:4]: All good.

**Review by richardtief — APPROVED** [review]
Thanks! Good to go

**Pull request merged**